packages/x-date-pickers/src/DateField/tests/selection.DateField.test.tsx

```diff
@@ -81,11 +86,12 @@ describe('<DateField /> - Selection', () => {
       // Simulate a touch focus interaction on mobile
       act(() => {
         input.focus();
-        input.setSelectionRange(9, 10);
-        clock.runToLast();
       });
-
+      clock.runToLast();
       expectInputValue(input, 'MM / DD / YYYY');
+
+      input.setSelectionRange(9, 11);
```

Comment:
Was this a mistake previously, that somehow did not fail the test? 🤔 🤷

Reply:
Correct, this test comes from #6207.

It seems that this test caught a regression, changing it hides it. See how on https://deploy-preview-8069--material-ui-x.netlify.app/x/react-date-pickers/fields/#fields-to-edit-a-single-element I need to tap twice to select the year:

https://user-images.githubusercontent.com/3165635/222933300-8d100831-fdf1-4207-a36b-2c8a6107f8df.mp4

Reply:
This behavior change is intended. It was a trade off between having a placeholder and being able to pick a section in a non focused input in a single click.

Reply:
Ah, OK, so using the <input> `placeholder` attribute over applying the placeholder color. This sounds simpler for developers when they provide a custom textbox.